src/DatePicker/YearButton.js

```diff
@@ -1,53 +1,35 @@
 import React from 'react';
 import EnhancedButton from '../internal/EnhancedButton';
 
-function getStyles(props, context) {
-  const {
-    selected,
-    year,
-  } = props;
-
-  const {hover} = context;
-
-  const {
-    baseTheme,
-    datePicker,
-  } = context.muiTheme;
+function getStyles(props, context, state) {
+  const {selected, year} = props;
+  const {baseTheme, datePicker} = context.muiTheme;
+  const {hover} = state;
 
   return {
     root: {
       boxSizing: 'border-box',
-      WebkitTapHighlightColor: 'rgba(0,0,0,0)', // Remove mobile color flashing (deprecated)
-      position: 'relative',
+      color: year === new Date().getFullYear() && datePicker.color,
       display: 'block',
-      margin: '0 auto',
-      width: 36,
       fontSize: 14,
-      padding: '8px 2px',
-      color: year === new Date().getFullYear() && datePicker.color,
+      margin: '0 auto',
+      position: 'relative',
+      textAlign: 'center',
+      WebkitTapHighlightColor: 'rgba(0,0,0,0)', // Remove mobile color flashing (deprecated)
     },
     label: {
+      alignSelf: 'center',
+      color: hover || selected ? datePicker.color : baseTheme.palette.textColor,
+      fontSize: selected ? 26 : 17,
+      fontWeight: hover ? 450 : selected ? 500 : 400,
```

Comment:
Yup. you guys are right. DO you guys want me to keep the font weight same for hover and selected? as its not that clear from the material spec if there is a separate hover font weight.

From what they have there it matches the 500 font weight for selected year.